src/errors_internal.rs

```diff
@@ -22,30 +25,41 @@ pub enum Error {
     /// An error indicating that a struct implementing the `PacketRouter` trait failed to handle a packet.
     #[error("Packet handler failed with error {source:?}")]
     PacketHandlerFailure {
+        /// The source error that caused the packet handler failure.
         source: Box<dyn std::error::Error + Send + Sync + 'static>,
     },
 
     /// An error indicating that the library failed to build a data stream within a stream builder utility function.
     #[error("Failed to build input data stream with error {source:?}")]
     StreamBuildError {
+        /// The source error that caused the stream build error.
         source: Box<dyn std::error::Error + Send + Sync + 'static>,
+        /// The description of the stream build error.
         description: String,
     },
 
     /// An error indicating that too much data is being sent.
+    /// NOTE: This error name is misspelled and should be treated as if it is `InvalidDataSize`.
     #[error("Trying to send too much data")]
-    InvalidaDataSize { data_length: usize },
+    InvalidaDataSize {
```

Comment:
This is unfortunate, but changing this value from `InvalidaDataSize` to `InvalidDataSize` would be a breaking change. We could either...

1. Do nothing (as I have done here)
2. Fix this and publish `v1.7.0`, introducing an insanely easy to fix breaking change in a minor version bump.
3. Wait until a `v2.0.0` goes out and include that change there

Reply:
Fixing a typo should not be a big deal, for sure we shouldn't wait for v2.0.0.

I'd like to make a new release in July, let's fix this and publish 1.7.0.